miniapp/miniapp_tsgemm.cpp

```diff
@@ -328,28 +331,43 @@ void sirius_gemm(ExecutorType const& mpi_executor, CommunicatorGrid& comm_grid,
       int tile_tag = dlaf::common::computeLinearIndex(Ordering::ColumnMajor, c_idx,
                                                       {tile_size.rows(), tile_size.cols()});
 
+      // Orders tasks so that the number of c_tiles computed in parallel at any moment is equal to the
+      // number of cores in the `default` thread pool.
+      int c_dep_i = tile_i - dep_tile_i;
+      int c_dep_j = tile_j - dep_tile_j;
```

Comment:
> We may evaluate if it is reasonable to add a constraint to SizeType, e.g. it has to be a signed integral type.

I'd be happy with `SizeType` as long as it's signed. If we decide to make it unsigned however it will not be a good fit for the loop because the logic allows and depends on `c_dep_i` and `c_dep_j` to be negative.

Reply:
SizeType is signed. It is meant to be the int type of BLAS and LAPACK.

Reply:
So, I would go for adding in `dlaf/types.h` next to `SizeType` definition the following constraint

```cpp
static_assert(std::is_integral<SizeType>::value && std::is_signed<SizeType>::value, "it has to be a signed integral type.");
```